tensorboard/data/server/cli/dynamic_logdir.rs

```diff
@@ -52,29 +67,43 @@ impl DynLogdir {
     /// [seanmonstar/reqwest#1017].
     ///
     /// [seanmonstar/reqwest#1017]: https://github.com/seanmonstar/reqwest/issues/1017
-    pub fn new(path: PathBuf) -> Option<Self> {
+    pub fn new(path: PathBuf) -> Result<Self, Error> {
         let path_str = path.to_string_lossy();
-        let gcs_path = match path_str.strip_prefix("gs://") {
-            // Assume that anything not starting with `gs://` is a path on disk.
-            None => return Some(DynLogdir::Disk(DiskLogdir::new(path))),
-            Some(p) => p,
-        };
+        if path_str.is_empty() {
+            return Err(Error::EmptyLogdir);
+        }
+
+        let protocol_parts: Vec<&str> = path_str.splitn(2, "://").collect();
```

Comment:
Heh. I feel like every time I look an an API and think it could be improved, I go looking and find out it's there but only [in Rust nightly](https://doc.rust-lang.org/std/string/struct.String.html#method.split_once).